sia/host.go

```diff
@@ -10,8 +11,22 @@ func (c *Core) HostInfo() (components.HostInfo, error) {
 
 // TODO: Make a better UpdateHost thing.
 func (c *Core) UpdateHost(announcement components.HostAnnouncement) error {
+	info, err := c.host.HostInfo()
+	if err != nil {
+		return err
+	}
+	// TODO: This stuff should not happen here, it should be managed by
+	// hostSetConfigHandler.
+	announcementUpdate := info.Announcement
+	announcementUpdate.IPAddress = c.server.Address()
+	announcementUpdate.TotalStorage = announcement.TotalStorage
+	announcementUpdate.MaxFilesize = announcement.MaxFilesize
+	announcementUpdate.MinTolerance = announcement.MinTolerance
+	announcementUpdate.Price = announcement.Price
+	announcementUpdate.Burn = announcement.Burn
```

Comment:
If I understand this correctly: in order to update the host, you need to pass in a `HostAnnouncement`. If you only want to change certain fields, you need to first obtain the old `HostAnnouncement`, modify it, then pass it back in. `UpdateHost` then *also* grabs the old `HostAnnouncement`, but it only modifies certain fields (the ones listed above). Then it sticks this into a different `HostUpdate` object, only one field of which is modified, and finally passes this object to a different `UpdateHost` function, which happens to share the same name despite taking a completely different argument.